internal/autoevent/executor.go

```diff
@@ -47,93 +48,71 @@ func (e *Executor) Run(ctx context.Context, wg *sync.WaitGroup, dic *di.Containe
-				if event.Origin == 0 {
-					event.Origin = common.GetUniqueOrigin()
-				}
-
 				// After the auto event executes a read command, it will create a goroutine to send out events.
 				// When the concurrent auto event amount becomes large, core-data might be hard to handle so many HTTP requests at the same time.
 				// The device service will get some network errors like EOF or Connection reset by peer.
 				// By adding a buffer here, the user can use the Service.AsyncBufferSize configuration to control the goroutine for sending events.
 				go func() {
-					m.autoeventBuffer <- true
-					common.SendEvent(event, lc, container.CoredataEventClientFrom(dic.Get))
-					<-m.autoeventBuffer
+					buffer <- true
+					// TODO: uncomment when v2 client refactoring is finished
+					//common.SendEvent(event, lc, container.CoredataEventClientFrom(dic.Get))
+					<-buffer
 				}()
 			} else {
-				lc.Debug(fmt.Sprintf("AutoEvent - no event generated when reading resource %s", e.autoEvent.Resource))
+				lc.Debugf("AutoEvent - no event generated when reading resource %s", e.resource)
 			}
 		}
 	}
 }
 
-func readResource(e *Executor, dic *di.Container) (*dsModels.Event, common.AppError) {
+func readResource(e *Executor, dic *di.Container) (event *dtos.Event, err errors.EdgeX) {
 	vars := make(map[string]string, 2)
 	vars[common.NameVar] = e.deviceName
-	vars[common.CommandVar] = e.autoEvent.Resource
+	vars[common.CommandVar] = e.resource
 
-	evt, appErr := handler.CommandHandler(vars, "", common.GetCmdMethod, "", dic)
-	return evt, appErr
+	res, err := application.CommandHandler(true, false, "", vars, "", dic)
+	if err != nil {
+		return event, err
+	}
+	return &res.Event, nil
 }
 
-func compareReadings(e *Executor, readings []contract.Reading, hasBinary bool, lc logger.LoggingClient) bool {
-	var identical bool = true
+func (e *Executor) compareReadings(reading dtos.BaseReading) bool {
 	e.rwMutex.RLock()
 	defer e.rwMutex.RUnlock()
-	for _, r := range readings {
-		switch e.lastReadings[r.Name].(type) {
-		case uint64:
-			checksum := xxhash.Checksum64(r.BinaryValue)
-			if e.lastReadings[r.Name] != checksum {
-				e.lastReadings[r.Name] = checksum
-				identical = false
-			}
-		case string:
-			v, ok := e.lastReadings[r.Name]
-			if !ok || v != r.Value {
-				e.lastReadings[r.Name] = r.Value
-				identical = false
-			}
-		case nil:
-			if hasBinary && len(r.BinaryValue) > 0 {
-				e.lastReadings[r.Name] = xxhash.Checksum64(r.BinaryValue)
-			} else {
-				e.lastReadings[r.Name] = r.Value
-			}
-			identical = false
-		default:
-			lc.Error("Error: unsupported reading type (%T) in autoevent - %v\n", e.lastReadings[r.Name], e.autoEvent)
-			identical = false
+
+	if reading.Value != "" {
+		if reading.Value != e.lastReadings.Value {
+			e.lastReadings = reading
+			return false
+		}
+		return true
+	} else {
+		if bytes.Compare(e.lastReadings.BinaryValue, reading.BinaryValue) != 0 {
```

Comment:
consider to use checksum for performance improvement

Reply:
after doing some benchmark, in worst case using `bytes.Compare` is ~0.7ns slower than calculating the checksum with 1GB slice. Do you think we still need the checksum here ?

Reply:
sounds good.  it looks like we don't need the checksum anymore.